### ISSUE TITLE
In our new report, the Ethiopia widgets are getting squashed.

### ISSUE THREAD

**Issue by richardbarad** [wri/gfw-mapbuilder#495]
In our new report, the Ethiopia widgets are getting squashed:

http://alpha.blueraster.io/gfw-mapbuilder/new-report-consolidated-builds/ethiopia/

If I run the Land Cover Analysis in the layer panel the widget looks like this and has a height of 225px.

![image](https://user-images.githubusercontent.com/40440743/62808260-9b663080-bac5-11e9-9a4e-3ccc06d9a8a0.png)

However, in the report the same widget get's squashed down to have a height of only 82px. 

![image](https://user-images.githubusercontent.com/40440743/62808515-45de5380-bac6-11e9-8ecb-e3065f6f27b0.png)

Is this something related to our resize signal? It looks like the widget should automatically include `"autosize":"pad",` setting which is what causes the widgets to adjust to be larger than the height specified in the config. Do we know why this is not happening in the report? 

https://vega.github.io/vega/docs/specification/#autosize

**Comment by KaylaKremer** [comment]
Hi @richardbarad ,

For the `autosize`, I'm actually using `fit` and not `pad`. 

Regarding the `height` issue, I think I have an idea as to what is happening. If you recall, we had only specified an override signal on the `width`. We didn't want the height to automatically adjust to fit its container div and stay fixed. (I am using the `Land Use-Land Cover` widget for my screenshot examples).

Width signal:
<img width="498" alt="Screen Shot 2019-08-12 at 10 51 54 AM" src="https://user-images.githubusercontent.com/22095244/62874532-6806ea00-bcef-11e9-8765-da02d1666759.png">

It looks like for both in the `analysis panel` as well as in the `report`, the widgets have fixed heights. For example, `Land Use-Land Cover` has a height of 75.

API Call:
![Screen Shot 2019-08-12 at 10 11 05 AM](https://user-images.githubusercontent.com/22095244/62874699-ba480b00-bcef-11e9-83ce-950cd7f50c31.png)

Analysis Panel:
![Screen Shot 2019-08-12 at 10 16 11 AM](https://user-images.githubusercontent.com/22095244/62874721-c2a04600-bcef-11e9-979e-88dfab8ccddf.png)

Report:
![Screen Shot 2019-08-12 at 10 16 39 AM](https://user-images.githubusercontent.com/22095244/62874758-ca5fea80-bcef-11e9-8ab6-3893c038d444.png)

For my auto resize to work, it uses `React Measure` along with the widget signal override, which is only used in the `report`, not the `analysis panel`. And it only updates the `width` to make that responsive, not the `height`. So it gets its height value from the widget's config, which for most of the Ethiopia widgets, isn't very high (in this `Land Use-Land Cover` example,  recall it's `75`).

When the widget renders in the analysis panel, it is not using `React Measure`. You'll notice that the width and height here appear to be fixed at specific values, it is not responsive:

<img width="639" alt="Screen Shot 2019-08-12 at 10 57 48 AM" src="https://user-images.githubusercontent.com/22095244/62875210-99cc8080-bcf0-11e9-85d4-c91f210087e8.png">

So basically, the original config for these widgets are very short in height to begin with. How they are in the `report` is their original height. In the `analysis panel`, the width and height is being set at certain values. I hope this makes sense, but it appears that everything is working as it should. If the Ethiopia widgets should have a higher height, then all that should be needed is to update their configs.

**Comment by richardbarad** [comment]
@KaylaKremer. Thanks! I think I see what is going on. Will test a few things tomorrow.

Richard

**Comment by richardbarad** [comment]
@KaylaKremer. I figured out the issue here.

If we look at the [documentation](https://vega.github.io/vega/docs/specification/) for the `autosize`, setting it to `fit` automatically adjust the layout in an attempt to force the total visualization size to fit within the given width, height and padding values. Because the width was set to `75` the widget content was getting squashed in order to fit the widget, axis, and labels within the 75px space allocated to the widget.

However, when the widget rendered in the layer panel `autosize` was set to `pad` which automatically increase the size of the view such that all visualization content is visible. This is the default `autosize` setting, and ensures that axes, legends and other items outside the normal width and height are included. 

Because this setting is the default, and our widget configurations did not contain `autosize` they were rendering withe `autosize` set to `pad`, and thus 75px was allocated to the actual widget and the widget was resizing to have a height high enough to fit the actual marks of the widget as well as the other content (legend, axis, etc.). To address this, I had to add `"autosize":"fit"` to the configs and also increase the heights. I have udpated all the configs for the Ethiopia widgets to reflect this.

However, I am wondering if we should standardize this behavior so that the `autosize` behavior is consistent in both the layer panel & the report. Do you recall why we decided to set `"autosize":"fit"` in the report?

**Comment by KaylaKremer** [comment]
@richardbarad ,

Yes, that all makes sense!

And I think it would be a good idea to standardize `autosize` in both the anlaysis and the report. Since right now in the code for the report, I'm having to manually adding it in anyways to the signal. I think it would be good if for all the charts, `autosize` was added to the signal so that we wouldn't have to worry about adding this in via code.

The reason for going with `"autosize":"fit"` versus the default `"autosize":"pad"` is that `React Measure` will not behave properly with `pad`. The extra padded values I think messes up how it calculates the size of the div container that holds the chart. `fit` was the only option that seemed to behave with `React Measure` to make the width responsive!

**Comment by richardbarad** [comment]
@KaylaKremer. Sounds like a good plan! Let's move forward with updating the signal then to always use `"autosize":"fit"`. Is this quick or would you like me to open a new ticket for this?

**Comment by richardbarad** [comment]
@KaylaKremer. Quick clarification here. In your comment above which were you suggesting:

1) Adding `"autosize":"fit"` on your side when the widget renders in the analysis panel is also uses fit

2) Adding `"autosize":"fit"` in all our Vega widget configurations in the API

**Comment by KaylaKremer** [comment]
@richardbarad ,

I was suggesting #2, to add it to the widget config in the API!

**Comment by richardbarad** [comment]
@KaylaKremer. Will do! We can close out this ticket.
